samples/boards/reel_board/mesh_badge/src/mesh.c

```diff
@@ -76,7 +78,67 @@ BT_MESH_HB_CB_DEFINE(hb_cb) = {
 	.recv = heartbeat,
 };
 
+static void cfg_cli_app_key_status_cb(struct bt_mesh_cfg_cli *cli, uint16_t
+				addr, uint8_t status, uint16_t net_idx,
+				uint16_t app_idx)
+{
+	struct bt_mesh_cfg_cli_mod_pub pub = {
+		.addr = GROUP_ADDR,
+		.app_idx = APP_IDX,
+		.ttl = DEFAULT_TTL,
+		.period = BT_MESH_PUB_PERIOD_SEC(10),
+	};
+	int err;
+
+	/* Bind to vendor model */
+	err = bt_mesh_cfg_cli_mod_app_bind_vnd(NET_IDX, addr, addr, APP_IDX,
+				MOD_LF, BT_COMP_ID_LF, NULL);
+	if (err) {
+		printk("Failed to bind to vendor model (err %d)", err);
+	}
```

Comment:
Perhaps in case of failure, you could set some boolean variable to false and then pass it to `show_mesh_status`. The rest configuration could be omitted in this case:
```
	bool res = true;

	...

	if (err) {
		printk("Failed to bind to vendor model (err %d)", err);
		res = false;
		goto end;
	}

	...

	printk("Configuration complete\n");
end:
	show_mesh_status(res, addr);
```